components/automate-ui/src/app/pages/project/rules/project-rules.component.ts

```diff
@@ -0,0 +1,193 @@
+import { GetRule, GetRules, CreateRule, UpdateRule } from 'app/entities/rules/rule.actions';
+import { getStatus, updateStatus, ruleFromRoute } from 'app/entities/rules/rule.selectors';
+import { projectFromRoute } from 'app/entities/projects/project.selectors';
+import { Project } from 'app/entities/projects/project.model';
+import { GetProject } from 'app/entities/projects/project.actions';
+
+@Component({
+  selector: 'app-project-rules',
+  templateUrl: './project-rules.component.html',
+  styleUrls: ['./project-rules.component.scss']
+})
+export class ProjectRulesComponent implements OnInit, OnDestroy {
+  public project: Project;
+  public ruleId: string;
+  public ruleForm: FormGroup;
+  public rule: Rule = <Rule>{};
+  public conditions: any[];
+  public isLoading = true;
+  public saving = false;
+  private isDestroyed: Subject<boolean> = new Subject<boolean>();
+
+  constructor(
+    private store: Store<NgrxStateAtom>,
+    private router: Router,
+    private route: ActivatedRoute,
+    private fb: FormBuilder) {
+
+      combineLatest(
+        this.store.select(getStatus),
+        this.store.select(updateStatus)
+      ).pipe(
+        takeUntil(this.isDestroyed),
+        map(([gStatus, uStatus]) => {
+          const routeId = this.route.snapshot.paramMap.get('ruleid');
+          this.isLoading =
+            routeId
+            ? (gStatus !== EntityStatus.loadingSuccess) ||
+              (uStatus === EntityStatus.loading)
+            : false;
+        })).subscribe();
+
+      this.store.select(routeParams).pipe(
+        pluck('ruleid'),
+        filter(identity),
+        takeUntil(this.isDestroyed))
+        .subscribe((id: string) => {
+          this.store.dispatch(new GetRule({ id }));
+        });
+
+      this.store.select(routeParams).pipe(
+        pluck('id'),
+        filter(identity),
+        takeUntil(this.isDestroyed))
+        .subscribe((id: string) => {
+          this.store.dispatch(new GetProject({ id }));
+        });
+
+      this.store.select(ruleFromRoute).pipe(
+        filter(identity),
+        takeUntil(this.isDestroyed),
+        map((state) => {
+          this.rule = <Rule>Object.assign({}, state);
+        })
+        ).subscribe();
+
+      this.store.select(projectFromRoute).pipe(
+        filter(identity),
+        takeUntil(this.isDestroyed),
+        map((state) => {
+          this.project = <Project>Object.assign({}, state);
+        })
+        ).subscribe();
+  }
+
+  ngOnInit(): void {
+    this.ruleForm = this.fb.group({
+      name: [this.rule.name || '', Validators.required],
+      type: [this.rule.type || '', Validators.required],
+      conditions: this.fb.array(this.populateConditions())
+    });
+  }
+
+  ngOnDestroy() {
+    this.isDestroyed.next(true);
+    this.isDestroyed.complete();
+  }
+
+  getHeading() {
+    return (this.ruleForm.value.name || !this.project)
+      ? `${this.ruleForm.value.name}`
+      : `${this.project.name}: Rule`;
```

Comment:
You might not have a project for a tiny fraction of time, but I don't think that needs special casing here.
More significantly, the project name should always appear.
Taken together, getHeading should just return this:
```
return `${this.project.name}: ` + (this.ruleForm.value.name || "Rule")
```

Reply:
@msorens The requirements is:
On create the heading should be <Project Name>: Rule, but once they have entered characters in the rule name it should dynamically change to update Rule to match the rule's name.

Reply:
@msorens I misread the requirement, thanks for the catch!